### PR TITLE
Pitch pages

### PR DESCRIPTION
**Pull request by mkosowsk**
This is a WIP PR and is not intended to be merged, just for guidance on the direction of the work 👌🏻

@thelostone-mc :)

TODO:

0. Confirm with Gitcoin core that Disqus is appropriate for the vision of Gitcoin and then integrate Disqus into these views, especially the detail view which is a skeleton at the moment. Disqus has a basic plan and also a plan that has up to 50,000 views for $9/month so seems pretty good 👍 
1. I made a migration here for the simple 'Idea'. Want to confirm that a web/app/retail/models.py should be created a la web/app/dashboard/models.py and then create this
2. Make reviews responsible for mobile/tablet
3. Have email sent when idea is submitted (settings.CONTACT_EMAIL)
4. Slackbot post as well

Thanks in advance for the feedback 💪🏻🤠👍🏻

### CODE REVIEW

**Comment by mkosowsk** [comment]
@mbeacom  if you have some extra cycles, please take a look to make sure I'm on the right path and not veering into the weeds 😂 👍 

Look to have this shipped within next 6 days!

**Comment by owocki** [comment]
> Confirm with Gitcoin core that Disqus is appropriate for the vision of Gitcoin and then integrate Disqus into these views, especially the detail view which is a skeleton at the moment. Disqus has a basic plan and also a plan that has up to 50,000 views for $9/month so seems pretty good 👍

I think so.. its better than building commenting from scratch.. do you disagree (or see the user case for a django-commenting system @mbeacom ) ? 

> I made a migration here for the simple 'Idea'. Want to confirm that a web/app/retail/models.py should be created a la web/app/dashboard/models.py and then create this

Makse sense to me.

File: app/assets/v2/css/ideas.css [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    -ms-transition:     border-color 0.25s;
+    -o-transition:      border-color 0.25s;
+    transition:         border-color 0.25s;
+}
+
+.idea-list:hover {
+	border-color: purple;
+}
+
+.idea-title {
```
Comment (owocki) [inline app/assets/v2/css/ideas.css]: this isnt in base.css! 🎉

File: app/retail/views.py [hunk 2]
```diff
@@ -463,6 +463,63 @@ def browser_extension_chrome(request):
...
+    #     'keywords': json.dumps([str(key) for key in Keyword.objects.all().values_list('keyword', flat=True)]),
+    # }
+    params = {}
+    return TemplateResponse(request, 'submit_idea.html', params)
+
+def ideas_list(request):
+    """TODO: MAKE A DESCRIPTIVE COMMENT"""
+    ideas = [
+        {
+            "title": "Find cheap, one-way flights within a departure data range",
```
Comment (owocki) [inline app/retail/views.py]: looking fwd to seeing the model here :)

**Comment by mkosowsk** [comment]
@owocki Thanks for taking a look, will open the final PR for this guy by this Sunday 3/25 👍  unfortunately work is eating my lunch at the moment so won't be able to take a look during the work week 😤

File: app/assets/v2/css/ideas.css [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	margin-bottom: 1.5em; 
+	border-style: solid; 
+	border-width: 1px; 
+	border-color: rgb(204, 204, 204); 
+	border-radius: 5px;
+	-webkit-transition: border-color 0.25s;
+    -moz-transition:    border-color 0.25s;
+    -ms-transition:     border-color 0.25s;
+    -o-transition:      border-color 0.25s;
+    transition:         border-color 0.25s;
```
Comment (thelostone-mc) [inline app/assets/v2/css/ideas.css]: Ah the indentation is off for the `transition`

File: app/assets/v2/css/ideas.css [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+.idea-list:hover {
+	border-color: purple;
```
Comment (thelostone-mc) [inline app/assets/v2/css/ideas.css]: Wanna swap it with the hexcode instead ?
Probably pick it up from @PixelantDesign style guide

File: app/assets/v2/css/ideas.css [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+.idea-title {
+	font-family: 'Muli', sans-serif; 
+	font-size: 1.15em; 
+	display: in-line
```
Comment (thelostone-mc) [inline app/assets/v2/css/ideas.css]: *`inline` :P

File: app/retail/templates/ideas_detail.html [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            
+            s.setAttribute('data-timestamp', +new Date());
+            (d.head || d.body).appendChild(s);
+        })();
+    </script>
+    {% include 'shared/bottom_notification.html' %}
+    {% include 'shared/analytics.html' %}
+    {% include 'shared/footer_scripts.html' %}
+    {% include 'shared/rollbar.html' %}
+    {% include 'shared/footer.html' %}
```
Comment (thelostone-mc) [inline app/retail/templates/ideas_detail.html]: Oh

File: app/retail/templates/ideas_detail.html [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+</body>
+
```
Comment (thelostone-mc) [inline app/retail/templates/ideas_detail.html]: Don't hate me for this 😅 
We are sticking to 2 spaces for HTML, CSS files
Also could you tab shift the whole `head` + `body` tag by one cause they are enclosed within the `html` tag

File: app/retail/templates/shared/head.html [hunk 8]
```diff
@@ -32,6 +32,7 @@
     <link rel="stylesheet" href="{% static "v2/css/timeline.css" %}" />
     <link rel="stylesheet" href="{% static "v2/css/carousel.css" %}" />
     <link rel="stylesheet" href="{% static "v2/css/faucet.css" %}" />
+    <link rel="stylesheet" href="{% static "v2/css/ideas.css" %}" />
```
Comment (thelostone-mc) [inline app/retail/templates/shared/head.html]: Thoughts on moving this to just the `ideas_list.html` page cause it won't be used anywhere else ? Would make the requests on the other pages a tad bit lighter

File: app/retail/templates/ideas_list.html [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    </div>
+    
+    
+    {% include 'shared/bottom_notification.html' %}
+    {% include 'shared/analytics.html' %}
+    {% include 'shared/footer_scripts.html' %}
+    {% include 'shared/rollbar.html' %}
+    {% include 'shared/footer.html' %}
+</body>
+
```
Comment (thelostone-mc) [inline app/retail/templates/ideas_list.html]: Same: Indentation as [above](https://github.com/gitcoinco/web/pull/647/files#r176520354)

File: app/retail/templates/shared/name.html [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+{% endcomment %}
+<div class="w-100 mt-2">
+    <label class="form__label" for="fullName">Full Name</label>
+    <input name='fullName' id='fullName'  class="w-100 form__input" type="text" placeholder="John Smith (optional)" value="{% if name %}{{name}}{% endif %}" />
```
Comment (thelostone-mc) [inline app/retail/templates/shared/name.html]: Same: Indentation as [above](https://github.com/gitcoinco/web/pull/647/files#r176520354)

File: app/retail/templates/submit_idea.html [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+                                        <div style="margin: 0.5rem">
+                                            <span id="summary_chars">0</span>/100
+                                        </div>
+                                    </div>
+                                    <textarea class="form__input" id="summary" name="summary" rows="1" maxlength="100" required>{{summary}}</textarea>
+                                </div>
+                                <div class="form-group">
+                                    <div class="justify-content-space-between">
+                                        <label class="form__label" for="more_info">Tell us more</label>
+                                        <div style="margin: 0.5rem">
```
Comment (thelostone-mc) [inline app/retail/templates/submit_idea.html]: Do away with the inline css ?

File: app/retail/templates/submit_idea.html [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+                                        {% else %}
+                                        {% endif %}
+                                        <input name="githubUsername" id="githubUsername" type=text class="form__input" placeholder="@you (optional)"   value="{% if github_handle %}{{ github_handle }}{% else %}{{githubUsername}}{% endif %}">
+                                    </div>
+                                </div>
+
+                                <div class="form-group">
+                                    <div class="justify-content-space-between">
+                                        <label class="form__label" for="summary">Summarize your idea</label>
+                                        <div style="margin: 0.5rem">
```
Comment (thelostone-mc) [inline app/retail/templates/submit_idea.html]: Do away with the inline css ?

File: app/retail/templates/submit_idea.html [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+          $('#summary_chars').text(length);
+        });
+
+        $('#more_info').keyup(function() {
+          var length = $(this).val().length;
+          $('#more_info_chars').text(length);
+        });
+    </script>  
+</body>
+
```
Comment (thelostone-mc) [inline app/retail/templates/submit_idea.html]: Same: Indentation as [above](https://github.com/gitcoinco/web/pull/647/files#r176520354)

**Comment by thelostone-mc** [comment]
@mkosowsk looks solid :D 
I'll pull in the code and test it out once you are done with the PR.
Left a few comments  but other than that we are looking good :D

**Comment by mkosowsk** [comment]
@thelostone-mc thanks as always for your help 👍 I will start to address in 24 hours time and look to have a PR turned around by the end of weekend!

File: app/retail/templates/shared/head.html (hunk 8 above)
Comment (mkosowsk) [reply app/retail/templates/shared/head.html]: I think ideas_detail may also eventually use stuff from v2/css/ideas.css 🤔

Comment (mkosowsk) [reply app/retail/templates/shared/head.html]: If ideas_detail.html doesn't use stuff from ideas.css I will yank this link out and drop in the ideas_list.html per your guidance 👍🏻

**Comment by owocki** [comment]
closing this for now as https://github.com/gitcoinco/web/pull/753 is the new active PR

**Pull request closed without merging**